keras_nlp/layers/transformer_decoder.py

```diff
@@ -84,6 +91,8 @@ def __init__(
         self.activation = activation
         self.layer_norm_epsilon = layer_norm_epsilon
         self._built = False
+        self.kernel_initializer = kernel_initializer
```

Comment:
I think you could actually call `kernel_initializer = keras.initializers.get(kernel_initializer)` here in init (same for bias). That would raise a friendly error message for a bad arg on layer creation, which I would find a lot more usable than only getting the error after calling the layer.

Reply:
We should probably actually do the same for activation while you are at it.

Reply:
I've made these changes for both of the initializers and the activation, Thanks!